improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -696,30 +709,28 @@ def linear_wet_bulb_fit(wet_bulb_temperature, heights, sea_points,
                 could be found, filled with zeros elsewhere.
 
         """
-        def fitting_function(wet_bulb_temps):
-            """
-            A small helper function used to find a linear fit of the
-            wet bulb temperature.
-            """
-            return linregress(
-                heights[start_point:end_point],
-                wet_bulb_temps[start_point:end_point])
         # Set up empty arrays for gradient and intercept
-        gradient = np.zeros(wet_bulb_temperature[0].shape)
-        intercept = np.zeros(wet_bulb_temperature[0].shape)
+        result_shape = wet_bulb_temperature.shape[1:]
+        gradient = np.zeros(result_shape)
+        intercept = np.zeros(result_shape)
         if np.any(sea_points):
+            # Use only subset of heights.
+            wbt = wet_bulb_temperature[start_point:end_point]
+            hgt = heights[start_point:end_point]
```

Comment:
probably not much point passing `start_point` and `end_point`, could as well pass `wbt` and `hgt` slices directly, or even `wet_bulb_temperature[start_point:end_point, sea_points]` and get rid off of another argument.

Reply:
Yep, that makes more sense